File: ArrayFunctions.java
import java.util.Scanner;

public class ArrayFunctions {

  public static int getCount(Integer[] arr) {
    int count = 0;
    for (Integer integer : arr) {
      if (integer != null) {
        count++;
      } else {
        break;
      }
    }
    return count;
  }

  public static void insert(Integer[] arr, int index, Integer value) {
    int count = getCount(arr);
    Integer temp;
    if (index < count) {
      for (int i = index; i <= count; i++) {
        temp = arr[i];
        arr[i] = value;
        value = temp;
      }
    } else if (index == count) {
      insert(arr, value);
    } else {
      System.out.println(
        "Error Can't Enter Value as Previous Locations are still null"
      );
    }
  }

  public static void insert(Integer[] arr, int value) {
    int count = getCount(arr);
    if (count < arr.length) {
      arr[count] = value;
    } else {
      System.out.println("ArrayOutOfBoundIndex");
    }
  }

  public static Integer pop(Integer[] arr, int index) {
    int count = getCount(arr);
    if (count == 0) {
      return -1;
    } else {
      Integer popItem = arr[index];
      for (int i = index; i < count; i++) {
        if (i == (count - 1)) {
          arr[i] = null;
          continue;
        }
        arr[i] = arr[i + 1];
      }
      return popItem;
    }
  }

  public static void update(Integer[] arr, int index, int value) {
    int count = getCount(arr);
    if (index >= arr.length) {
      System.out.println("index goes beyond array size");
    } else if (index >= count) {
      System.out.println("No Element exists at that index");
    } else {
      arr[index] = value;
    }
  }

  public static int search(Integer[] arr, int value) {
    int index = -1;
    int count = getCount(arr);
    for (int i = 0; i < count; i++) {
      if (value == arr[i]) {
        return i;
      } else {
        continue;
      }
    }
    return index;
  }

  public static void display(Integer[] array) {
    for (Integer integer : array) {
      System.out.print(integer + " ");
    }
    System.out.println("");
  }

  public static void main(String[] args) {
    Scanner ask = new Scanner(System.in);
    System.out.print("Enter the Size of Array: ");
    int size = ask.nextInt();
    Integer[] array = new Integer[size];
    insert(array, 45);
    insert(array, 54);
    insert(array, 64);
    insert(array, 97);
    display(array);
    System.out.println(search(array, 64));
    System.out.println(pop(array, 0));
    System.out.println(pop(array, 0));
    System.out.println(pop(array, 0));
    display(array);
  }
}

File: ArrayFunctions.py
class array:
    arr = []
    size = 0

    def __init__(self, size):
        self.size = size

    def insert(self,  value, index=len(arr)):
        if index >= self.size or len(self.arr) >= self.size:
            print("OverFlow!!")
            return False
        else:
            self.arr.insert(index, value)
        # self.arr[index] = value
        print(self.arr)
        return True

    def pop(self, index=(len(arr)-1)):
        if len(self.arr) == 0:
            print("UnderFlow!!")
            return False
        # print(f"i am  {index}")
        return self.arr.pop(index)

    def update(self, index, value):
        if len(self.arr) == 0:
            return False
        self.arr[index] = value
        return True

    def search(self, value):
        if len(self.arr) == 0:
            return False
        try:
            return self.arr.index(value)
        except ValueError:
            return -1

    def display(self):
        for i in range(0, len(self.arr)):
            print(self.arr[i], end=" ")
        print("")


myarr = array(5)
myarr.insert(4)
myarr.insert(4, 0)
myarr.insert(2, 1)
myarr.insert(3, 2)
myarr.insert(4, 3)
myarr.insert(5, 4)  # size exceed
myarr.insert(6, 5)  # size exceed
myarr.display()
print(myarr.pop())
print(myarr.pop(2))
print(myarr.search(45))
myarr.update(2, 600)
print(myarr.search(600))
myarr.display()
print(myarr.pop())
print(myarr.pop())
print(myarr.pop())
myarr.pop()

